pkg/builtin/multi/substring.go

```diff
@@ -0,0 +1,644 @@
+	{[]types.T{types.T_varchar, types.T_int16, types.T_int32}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int16, types.T_int64}, types.T_varchar},
+
+	{[]types.T{types.T_varchar, types.T_int32, types.T_uint8}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int32, types.T_uint16}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int32, types.T_uint32}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int32, types.T_uint64}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int32, types.T_int8}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int32, types.T_int16}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int32, types.T_int32}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int32, types.T_int64}, types.T_varchar},
+
+	{[]types.T{types.T_varchar, types.T_int64, types.T_uint8}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int64, types.T_uint16}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int64, types.T_uint32}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int64, types.T_uint64}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int64, types.T_int8}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int64, types.T_int16}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int64, types.T_int32}, types.T_varchar},
+	{[]types.T{types.T_varchar, types.T_int64, types.T_int64}, types.T_varchar},
+}
+
+func init() {
+	// function name registration
+	extend.FunctionRegistry["substring"] = builtin.Substring
+	extend.FunctionRegistry["substr"] = builtin.Substring
+
+	// append function parameter types and return types
+	for _, item := range argAndrets {
+		overload.AppendFunctionRets(builtin.Substring, item.args, item.ret)
+	}
+
+	// define a get return type function for substring function
+	extend.MultiReturnTypes[builtin.Substring] = func(extend []extend.Extend) types.T {
+		return getMultiReturnType(builtin.Substring, extend)
+	}
+
+	// define a stringify function for sbustring
+	extend.MultiStrings[builtin.Substring] = func(extends []extend.Extend) string {
+		if len(extends) == 2 {
+			return fmt.Sprintf("substring(%s, %s)", extends[0], extends[1])
+		} else {
+			return fmt.Sprintf("substring(%s, %s, %s)", extends[0], extends[1], extends[2])
+		}
+	}
+
+	// register subtring function type
+	overload.OpTypes[builtin.Substring] = overload.Multi
+
+	// Preparation for function calling
+	overload.MultiOps[builtin.Substring] = []*overload.MultiOp{
+		//----------------------------------------process char parameter--------------------------------------
+		{
+			Min:        2,
+			Max:        3,
+			Typ:        types.T_char,
+			ReturnType: types.T_char,
+			Fn: func(inputVecs []*vector.Vector, proc *process.Process, cs []bool) (*vector.Vector, error) {
+				// get the number of substr function parameters
+				var paramNum int = len(inputVecs)
+				column_src_col := inputVecs[0].Col.(*types.Bytes)
+				var column_start_const bool = cs[1]
```

Comment:
to consistent with our repo's naming convention, I think CamelCase would be better

Reply:
ok， I will modify it with CamelCase